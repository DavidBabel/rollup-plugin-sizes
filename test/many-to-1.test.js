"use strict";

var rollup  = require("rollup").rollup,
    
    plugin = require("../index.js");

describe("rollup-plugin-sizes", () => {
    describe("multiple-entry, single-output bundle", () => {
        var { log, out } = require("./log.js")();

        it("should show basic output", () =>
            rollup({
                input : [
                    "./test/specimens/many-to-1/a.js",
                    "./test/specimens/many-to-1/c.js",
                ],

                experimentalCodeSplitting : true,

                plugins : [
                    plugin({
                        log
                    })
                ]
            })
            .then((bundle) => bundle.generate({ format : "es" }))
            .then(() => expect(out()).toMatchSnapshot())
        );

        it("should show detailed output ", () =>
            rollup({
                input : [
                    "./test/specimens/many-to-1/a.js",
                    "./test/specimens/many-to-1/c.js",
                ],

                experimentalCodeSplitting : true,

                plugins : [
                    plugin({
                        log,

                        details : true
                    })
                ]
            })
            .then((bundle) => bundle.generate({ format : "es" }))
            .then(() => expect(out()).toMatchSnapshot())
        );
    });

    describe("multiple-entry, single-output bundle w/ dynamic imports", () => {
        var { log, out } = require("./log.js")();

        it("should show basic output", () =>
            rollup({
                input : [
                    "./test/specimens/many-to-1/a.js",
                    "./test/specimens/many-to-1/e.js",
                ],

                experimentalCodeSplitting : true,
                experimentalDynamicImport : true,

                plugins : [
                    plugin({
                        log
                    })
                ]
            })
            .then((bundle) => bundle.generate({ format : "es" }))
            .then(() => expect(out()).toMatchSnapshot())
        );

        it("should show detailed output ", () =>
            rollup({
                input : [
                    "./test/specimens/many-to-1/a.js",
                    "./test/specimens/many-to-1/e.js",
                ],

                experimentalCodeSplitting : true,
                experimentalDynamicImport : true,

                plugins : [
                    plugin({
                        log,

                        details : true
                    })
                ]
            })
            .then((bundle) => bundle.generate({ format : "es" }))
            .then(() => expect(out()).toMatchSnapshot())
        );
    });
});
