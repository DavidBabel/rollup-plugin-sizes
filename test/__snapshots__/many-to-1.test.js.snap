// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rollup-plugin-sizes multiple-entry, single-output bundle should show basic output 1`] = `
"Details for \\"./chunk1.js:\\"
	./chunk1.js - 21 B (100.00%)
Details for \\"./a.js:\\"
	./a.js - 44 B (100.00%)
Details for \\"./c.js:\\"
	./c.js - 113 B (100.00%)
"
`;

exports[`rollup-plugin-sizes multiple-entry, single-output bundle should show detailed output  1`] = `
"Details for \\"./chunk1.js:\\"
	./chunk1.js - 21 B (100.00%)
		b.js - 21 B (100.00%)
Details for \\"./a.js:\\"
	./a.js - 44 B (100.00%)
		a.js - 44 B (100.00%)
Details for \\"./c.js:\\"
	./c.js - 113 B (100.00%)
		c.js - 92 B (81.42%)
		d.js - 21 B (18.58%)
"
`;

exports[`rollup-plugin-sizes multiple-entry, single-output bundle w/ dynamic imports should show basic output 1`] = `
"Details for \\"./chunk1.js:\\"
	./chunk1.js - 21 B (100.00%)
Details for \\"./a.js:\\"
	./a.js - 44 B (100.00%)
Details for \\"./e.js:\\"
	./e.js - 37 B (100.00%)
Details for \\"./c.js:\\"
	./c.js - 113 B (100.00%)
"
`;

exports[`rollup-plugin-sizes multiple-entry, single-output bundle w/ dynamic imports should show detailed output  1`] = `
"Details for \\"./chunk1.js:\\"
	./chunk1.js - 21 B (100.00%)
		b.js - 21 B (100.00%)
Details for \\"./a.js:\\"
	./a.js - 44 B (100.00%)
		a.js - 44 B (100.00%)
Details for \\"./e.js:\\"
	./e.js - 37 B (100.00%)
		e.js - 37 B (100.00%)
Details for \\"./c.js:\\"
	./c.js - 113 B (100.00%)
		c.js - 92 B (81.42%)
		d.js - 21 B (18.58%)
"
`;
